File: from-0901-to-1000/0938-range-sum-of-bst/solution.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
 func rangeSumBST(root *TreeNode, low int, high int) int {
    result := 0
    var dfs func(node TreeNode)
    dfs = func(node TreeNode) {
        if low <= node.Val && node.Val <= high {
            result += node.Val
        } 
        if node.Val > low && node.Left != nil {
            dfs(*node.Left)
        } 
        if node.Val < high && node.Right != nil {
            dfs(*node.Right)
        }
    }
    dfs(*root)
    return result
}

File: from-0901-to-1000/0938-range-sum-of-bst/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        def dfs(node: Optional[TreeNode]) -> None:
            if not node:
                return
            nonlocal result, low, high
            if low <= node.val <= high:
                result += node.val
            if node.val > low:
                dfs(node.left)
            if node.val < high:
                dfs(node.right)

        result = 0
        dfs(root)
        return result
            